torch/csrc/distributed/c10d/Functional.cpp

```diff
@@ -422,6 +422,119 @@ at::Tensor all_to_all_single_autograd(
       input, output_split_sizes, input_split_sizes, group_name)[0];
 }
 
+class ReduceScatterTensor
+    : public torch::autograd::Function<ReduceScatterTensor> {
+ public:
+  static torch::autograd::Variable forward(
+      torch::autograd::AutogradContext* ctx,
+      const at::Tensor& input,
+      std::string reduce_op,
+      int64_t group_size,
+      std::string group_name) {
+    TORCH_CHECK(reduce_op == "sum", "Only sum reduce op is supported");
+
+    ctx->saved_data["group_size"] = group_size;
+    ctx->saved_data["group_name"] = group_name;
+
+    return c10::Dispatcher::singleton()
+        .findSchemaOrThrow("_c10d_functional::reduce_scatter_tensor", "")
+        .typed<decltype(reduce_scatter_tensor)>()
+        .call(input, reduce_op, group_size, group_name);
+  }
+
+  static torch::autograd::variable_list backward(
+      torch::autograd::AutogradContext* ctx,
+      torch::autograd::variable_list grad_out_list) {
+    const int64_t group_size = ctx->saved_data["group_size"].toInt();
+    const std::string& group_name = ctx->saved_data["group_name"].toStringRef();
+
+    DCHECK(grad_out_list.size() == 1);
+    auto grad_out = grad_out_list[0];
+
+    auto out =
+        c10::Dispatcher::singleton()
```

Comment:
can't we just call `all_gather_into_tensor` here? or the purpose of this is to basically make sure it goes through dispatcher again? if it's the later case, we should call `_c10d_functional_autograd` instead?